pandas/core/strings/accessor.py

```diff
@@ -1540,10 +1556,21 @@ def replace(
         if case is None:
             case = True
 
-        result = self._data.array._str_replace(
-            pat, repl, n=n, case=case, flags=flags, regex=regex
-        )
-        return self._wrap_result(result)
+        if isinstance(pat, dict):
+            res_output = self._data
+            for key, value in pat.items():
+                result = res_output.array._str_replace(
+                    key, str(value), n=n, case=case, flags=flags, regex=regex
```

Comment:
Why `str(value)` here? This introduces inconsistencies between dict vs non-dict paths I think.